WordPress/src/main/java/org/wordpress/android/ui/sitecreation/segments/NewSiteCreationSegmentsViewModel.kt

```diff
@@ -14,21 +15,26 @@ import org.wordpress.android.fluxc.Dispatcher
 import org.wordpress.android.fluxc.model.vertical.VerticalSegmentModel
 import org.wordpress.android.fluxc.store.VerticalStore.OnSegmentsFetched
 import org.wordpress.android.models.networkresource.ListState
-import org.wordpress.android.models.networkresource.ListState.Error
 import org.wordpress.android.models.networkresource.ListState.Loading
 import org.wordpress.android.modules.IO_DISPATCHER
 import org.wordpress.android.modules.MAIN_DISPATCHER
-import org.wordpress.android.ui.sitecreation.segments.NewSiteCreationSegmentsViewModel.ItemUiState.HeaderUiState
-import org.wordpress.android.ui.sitecreation.segments.NewSiteCreationSegmentsViewModel.ItemUiState.ProgressUiState
-import org.wordpress.android.ui.sitecreation.segments.NewSiteCreationSegmentsViewModel.ItemUiState.SegmentUiState
+import org.wordpress.android.ui.sitecreation.segments.SegmentsItemUiState.HeaderUiState
+import org.wordpress.android.ui.sitecreation.segments.SegmentsItemUiState.ProgressUiState
+import org.wordpress.android.ui.sitecreation.segments.SegmentsItemUiState.SegmentUiState
+import org.wordpress.android.ui.sitecreation.segments.SegmentsUiState.SegmentsContentUiState
+import org.wordpress.android.ui.sitecreation.segments.SegmentsUiState.SegmentsErrorUiState
 import org.wordpress.android.ui.sitecreation.usecases.FetchSegmentsUseCase
+import org.wordpress.android.util.NetworkUtilsWrapper
 import org.wordpress.android.viewmodel.SingleLiveEvent
 import javax.inject.Inject
 import javax.inject.Named
 import kotlin.coroutines.experimental.CoroutineContext
 
+private const val SHOW_LOADING_TO_PROVIDE_FEEDBACK_ARTIFICIAL_DELAY = 1000
```

Comment:
Maybe `CONNECTION_ERROR_DELAY_TO_SHOW_LOADING_STATE_FOR_FEEDBACK`, if it's too long `CONNECTION_ERROR_DELAY_TO_SHOW_LOADING_STATE` could also work.

I am totally fine with the current one as well, it's much better than fake delay 🙈

Reply:
👍